### ISSUE TITLE
[BUG] Internal Server Error

### ISSUE THREAD

**Issue by luwol03** [inventree/InvenTree#3355]
**Describe the bug**
I get an Internal Server error after start and don't know why.

**Steps to Reproduce**

1. Start the server
2. Watch logs
3. Navigate to `/admin`

**Expected behavior**
Should show the Django admin

![](https://user-images.githubusercontent.com/60048565/179567211-ee9312ce-17bd-46b1-9523-12cc2e612230.png)

**Deployment Method**
- [x] Docker
- [ ] Bare Metal

**Version Information**

```
# Version Information:
InvenTree-Version: 0.7.5
Django Version: 3.2.13


Database: postgresql
Debug-Mode: False
Deployed using Docker: True
```

Log:
https://gist.github.com/luwol03/45794e822bfd00730f9edcb259028915

**Comment by matmair** [comment]
I can not reproduce this with a fresh install @luwol03 anything else you changed? Do you have any custom plugins activated?

**Comment by SchrodingersGat** [comment]
@luwol03 this is a new one for me too. Looks like the postgres database container is having some issues. Please let us know if you made any changes to the `docker-compose` or `.env` files

**Comment by luwol03** [comment]
Hello, the first lines until line 826 are cannons connect to database, because the database is not listening, but then the connection is successful but an SQL statement is executed 3 times concurrently which causes an error. And I guess this has something to to with that the 500 on /admin. I more and more think, that this is caused because I run it via macos docker. (Also the git safepermissions error). I try to export the database and try on Linux. Thank you for taking a look.

**Comment by SchrodingersGat** [comment]
@luwol03 please let us know if you find anything further :)

**Comment by luwol03** [comment]
@SchrodingersGat I now copied everything to a linux machine and started the containers, open `/admin` in browser and see error. I have no idea why. I would appreciate if you could take a look. I bundled everything into a targz archive:[inventree-mac.tar.zip](https://github.com/inventree/InvenTree/files/9142308/inventree-mac.tar.zip)

### Steps to reproduce:
1. extract the tar from the zip (gh only allowed me to upload zip)
2. `tar xfzv inventree-mac.tar`
3. `cd inventree-mac`
4. `docker-compose up -d && docker-compose logs -f`
5. wait for start and see error
6. browse `http://localhost:8000/admin` (if login is needed, there is an admin user: `admin` with password: `Passw0rd`)

**Comment by luwol03** [comment]
Hey @SchrodingersGat, I analyzed it a bit further and I did the following:
1. Running the inventree_data volume with a clean database -> OK
2. Running my db -> ERROR
3. Comparing the dbs table by table
4. After trying to modify one by one of my corrupted db to fit the working (fresh) db I found the "Übeltäter". If `simpleactionplugin` is enabled, I get the error described in this issue. Removing the checkmark and restarting the docker container results with no error
   before: ![image](https://user-images.githubusercontent.com/60048565/179843712-fdda41d7-9f59-493c-992b-524d4af97e00.png)
   after: ![image](https://user-images.githubusercontent.com/60048565/179844281-4ce8abdf-80c8-4126-b0f7-3e94ad3ae87e.png)


### Excact descibed Error:

<details>

![image](https://user-images.githubusercontent.com/60048565/179844366-43caa70f-6d23-47c1-803d-41b539cd1470.png)

```bash
Environment:


Request Method: GET
Request URL: http://localhost:8000/admin/

Django Version: 3.2.13
Python Version: 3.9.5
Installed Applications:
['build.apps.BuildConfig',
 'common.apps.CommonConfig',
 'company.apps.CompanyConfig',
 'label.apps.LabelConfig',
 'order.apps.OrderConfig',
 'part.apps.PartConfig',
 'report.apps.ReportConfig',
 'stock.apps.StockConfig',
 'users.apps.UsersConfig',
 'plugin.apps.PluginAppConfig',
 'InvenTree.apps.InvenTreeConfig',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'user_sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'maintenance_mode',
 'django_filters',
 'rest_framework',
 'rest_framework.authtoken',
 'corsheaders',
 'crispy_forms',
 'import_export',
 'django_cleanup.apps.CleanupConfig',
 'mptt',
 'markdownx',
 'markdownify',
 'django_admin_shell',
 'djmoney',
 'djmoney.contrib.exchange',
 'error_report',
 'django_q',
 'formtools',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'django_otp',
 'django_otp.plugins.otp_totp',
 'django_otp.plugins.otp_static',
 'allauth_2fa',
 'sslserver']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'x_forwarded_for.middleware.XForwardedForMiddleware',
 'user_sessions.middleware.SessionMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'InvenTree.middleware.InvenTreeRemoteUserMiddleware',
 'django_otp.middleware.OTPMiddleware',
 'InvenTree.middleware.CustomAllauthTwoFactorMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'InvenTree.middleware.AuthRequiredMiddleware',
 'InvenTree.middleware.Check2FAMiddleware',
 'maintenance_mode.middleware.MaintenanceModeMiddleware',
 'error_report.middleware.ExceptionProcessor']



Traceback (most recent call last):
  File "/home/inventree/.local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/contrib/admin/sites.py", line 250, in wrapper
    return self.admin_view(view, cacheable)(*args, **kwargs)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/utils/decorators.py", line 130, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/contrib/admin/sites.py", line 232, in inner
    return view(request, *args, **kwargs)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/contrib/admin/sites.py", line 522, in index
    app_list = self.get_app_list(request)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/contrib/admin/sites.py", line 505, in get_app_list
    app_dict = self._build_app_dict(request)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/contrib/admin/sites.py", line 487, in _build_app_dict
    'app_url': reverse(
  File "/home/inventree/.local/lib/python3.9/site-packages/django/urls/base.py", line 86, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
  File "/home/inventree/.local/lib/python3.9/site-packages/django/urls/resolvers.py", line 698, in _reverse_with_prefix
    raise NoReverseMatch(msg)

Exception Type: NoReverseMatch at /admin/
Exception Value: Reverse for 'app_list' with keyword arguments '{'app_label': 'common'}' not found. 1 pattern(s) tried: ['admin/(?P<app_label>part|build|order|stock)/$']
```

</details>


Now my log looks like that (this time debug mode and DEBUG log level enabled) (take care to look at the right file, just appended that to the gist): https://gist.github.com/luwol03/45794e822bfd00730f9edcb259028915#file-new_log-log

As you can see, the main error is gone, but the concurrent update is still there. I think this should be tracked in a new issue, because this is unrelated to the 5XX error.

Hopefully you have an idea why an enabled `simpleactionplugin` causes that 5XX error.

luwol03

**Comment by SchrodingersGat** [comment]
> As you can see, the main error is gone, but the concurrent update is still there. I think this should be tracked in a new issue, because this is unrelated to the 5XX error.

This has actually been addressed for `0.8.0` - https://github.com/inventree/InvenTree/pull/3268

### Debug Mode

Did you change the "debug" configuration between runs? There is a discrepancy between the version information you posted, and the log:

![image](https://user-images.githubusercontent.com/10080325/179869678-734d91ee-2e7c-441b-865e-f829326c2108.png)

![image](https://user-images.githubusercontent.com/10080325/179869736-9ca60b80-190c-418d-880d-695ba58c4d98.png)

**Comment by SchrodingersGat** [comment]
There are some strange issues popping up with javascript URLs also

![image](https://user-images.githubusercontent.com/10080325/179875727-577689ef-87ea-4df3-951e-80c83bbb9b9e.png)

**Comment by SchrodingersGat** [comment]
@luwol03 I have managed to reproduce the **500** error when navigating to the admin page, with the zip you provided. Will look into it further.

**Comment by SchrodingersGat** [comment]
Next observation. I changed the target image from `inventree:stable` to `inventree:latest` and the problem persists. However it is particular to your setup files - I cannot replicate this on any of:

- local `dev` setup
- local `stable` production setup
- local `latest` production setup
- demo server

**Comment by luwol03** [comment]
@SchrodingersGat Have you read that I found out, that one single flag in the db is causing that?

**Comment by SchrodingersGat** [comment]
> Have you read that I found out, that one single flag in the db is causing that?

I did read that, I am very confused as to why that would lead to all these errors..

**Comment by SchrodingersGat** [comment]
Is it otherwise working well for you now?

**Comment by luwol03** [comment]
> Is it otherwise working well for you now?

Yes, inventree rocks, why do I don't discovered that before. I was a long time searching for a free storage tracking solution but didn't found inventree via google. It was a randomness, that inventree was on my gh explore page. Good job.

How do you actually finance this project and the investigated time?

(I only had this 500 on my experimental instance hosted on my pc to test the plugins and write my own first plugin. Only thought this may be an error which could occur in someone else production instance and would be good to fix.)

**Comment by SchrodingersGat** [comment]
@luwol03 I think this may have been specific to your particular setup, it is not something we have seen previously. I'm going to close this out for now, but if you see it again, please comment back here.

**Comment by Zontex** [comment]
My issue #3476 seems to be duplicated. I also had this issue now after upgrading but after reading @luwol03  notes I disabled plugins and now it's working. is there any fix to enable plugins back?

**Comment by SchrodingersGat** [comment]
Huh, maybe one of the plugins is broken? Does the error come back if you re-enable plugins?

**Comment by Zontex** [comment]
> Huh, maybe one of the plugins is broken? Does the error come back if you re-enable plugins?

I disabled the default simpleactionplugin via MySQL by running "UPDATE plugin_pluginconfig SET active=0 WHERE id=1;" then re-enabled the plugins and it works

**Comment by Zontex** [comment]
So let's summarize it to enable understand better what's going on. @luwol03 didn't provide much information on how the problem occurs but he did find the solution which is related to the plugin. 

He was running a **Docker** while I was running **bare-metal** - which affects both.

For me, it happened after the 0.7.4 to 0.8.0 upgrade. I had plugins enabled in 0.7.4 but I never used them so the only plugin that was activated was "**simpleactionplugin**", by disabling it manually I was able to solve the issue. 

## TODO

- [ ] Enable another different plugin to see if the error is related to that specific plugin or the entire plugin system after the upgrade.

**Comment by matmair** [comment]
@Zontex I had problems reproducing this last time - maybe because of the plugin settings. We might need to include plugin info in the version information stuff.

**Comment by SchrodingersGat** [comment]
@Zontex are you able to provide a bit more detail here - at what exact point do you see the error?

Is it during the update process? e.g. running `invoke update` or similar command?

Or is it *after* a successful update procedure when you navigate to the website

**Comment by Zontex** [comment]
After successful installation once you try to access "admin/" page. Exactly
the same thing happened to OP

**Comment by SchrodingersGat** [comment]
@Zontex ah yep specific to the admin page. I had overlooked that detail. I can definitely recreate that.

**Comment by SchrodingersGat** [comment]
![Uploading image.png…]()

**Comment by Zontex** [comment]
@SchrodingersGat after updating will the default plugin work or still the admin dashboard won't function?

**Comment by SchrodingersGat** [comment]
@Zontex after the patch in #3484 it should all work perfectly. If you are able to test on `inventree:master` please do. Otherwise `0.8.1` will be released soon.

**Comment by SchrodingersGat** [comment]
@luwol03 please note that *finally* I think I have found a solution to this one.
